frontend/components/MainPage/header/AgentButton/AgentNotRunningButton.tsx

```diff
@@ -223,13 +223,34 @@ export const AgentNotRunningButton = () => {
       }
     }
 
-    if (isNil(service) && isNil(middlewareServiceResponse))
-      throw new Error('Service not found');
+    // Update if service already exists and template hash has changed
+    if (!createServiceResponse && service) {
+      try {
+        if (service.hash !== serviceTemplate.hash) {
```

Comment:
is this needed to support previous versions?

Reply:
correct, there were issues when users migrated from legacy staging to the latest; the template hashes had updated but did not support the new environment